engine/pkg/rpcerror/grpc.go

```diff
@@ -141,9 +141,16 @@ func UnaryServerInterceptor(ctx context.Context, req interface{}, info *grpc.Una
 	if err != nil {
 		errOut := ToGRPCError(err)
 		logger := log.With(zap.String("method", info.FullMethod), zap.Error(errOut), zap.Any("request", req))
-		if s, ok := status.FromError(errOut); ok && s.Code() == codes.Internal {
+		s, ok := status.FromError(errOut)
+		if !ok {
+			s = status.FromContextError(err)
+		}
+		switch s.Code() {
+		case codes.Unknown:
+			logger.Warn("request handled with an unknown error")
+		case codes.Internal:
 			logger.Warn("request handled with an internal error")
-		} else {
+		default:
 			logger.Debug("request handled with an error")
```

Comment:
Do we need to log more information on the request here?

Reply:
`logger` already contains the method, error, and request info. Do we still need more information here?